apps/server/src/controllers/admin/user.js

```
@@ -0,0 +1,39 @@
+import {
+  getUsers,
+} from '../../models/queries/user-queries.js'
+
+import { send200, send500 } from '../../utils/response.js'
+import { getLogInfos } from '../../utils/logger.js'
+import UnauthorizedError from '../../utils/exceptions.js'
+
+export const getUsersController = async (req, res) => {
```

Hello,
je pense que tu as moyen de bien simplifier tout ça !
on essaie de faire du fail fast : 

première ligne du try tu fais un `if (!req.session.user.groups.includes('admin')) throw new Error('Vous n'avez les droits administrateur')`
ensuite tu fais ton `getUsers`
et puis tu catch l'error et tu la renvoies telle quelle (pas besoin de classe)
:)

---

J'irais même un peu plus loin, avec quelque chose du style :
```js
try {
  if (! req.session.user.groups?.includes('admin')) {
    send500(res, 'Vous n\'avez pas les droits administrateurs')
    return
  }
  const users = await getUsers()
  req.log.info({
    ...getLogInfos(),
    description: 'Users successfully retreived',
  })
  send200(res, users)
} catch (error) {
  const message = 'Echec de récupération des utilisateurs'
  req.log.error({
    ...getLogInfos(),
    description: message,
    error: error.message,
  })
  send500(res, message)
}
```

Ce qui permet d'avoir l'erreur custom que tu voulais sans créer de classe et donc diminuer le nombre de lignes de code et globalement diminuer la complexité.